client/components/payment-activity/date-range-picker.tsx

```diff
@@ -0,0 +1,155 @@
+	},
+	last_7_days: {
+		start: now
+			.clone()
+			.subtract( 7, 'days' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 7 days', 'woocommerce-payments' ),
+	},
+	last_4_weeks: {
+		start: now
+			.clone()
+			.subtract( 4, 'weeks' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 4 weeks', 'woocommerce-payments' ),
+	},
+	last_3_months: {
+		start: now
+			.clone()
+			.subtract( 3, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 3 months', 'woocommerce-payments' ),
+	},
+	last_12_months: {
+		start: now
+			.clone()
+			.subtract( 12, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 12 months', 'woocommerce-payments' ),
+	},
+	month_to_date: {
+		start: now.clone().startOf( 'month' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Month to date', 'woocommerce-payments' ),
+	},
+	quarter_to_date: {
+		start: now.clone().startOf( 'quarter' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Quarter to date', 'woocommerce-payments' ),
+	},
+	year_to_date: {
+		start: now.clone().startOf( 'year' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Year to date', 'woocommerce-payments' ),
+	},
+};
+
+const formatDateRange = (
+	start: moment.Moment,
+	end: moment.Moment
+): string => {
+	// Today - show only today's date.
+	if ( start.isSame( end, 'day' ) ) {
+		return start.format( 'MMMM D, YYYY' );
+	}
+
+	// Different years - show year for both start and end
+	if ( ! start.isSame( end, 'year' ) ) {
+		return `${ start.format( 'MMMM D, YYYY' ) } - ${ end.format(
+			'MMMM D, YYYY'
+		) }`;
+	}
+
+	// Same year - show year only for end date.
+	return `${ start.format( 'MMMM D' ) } - ${ end.format( 'MMMM D, YYYY' ) }`;
+};
+
+const options = Object.keys( timeOptions ).map( ( optionKey ) => {
+	const option = timeOptions[ optionKey ];
+	return {
+		key: optionKey,
+		name: option.displayKey,
+		hint: formatDateRange( option.start, option.end ),
+	};
+} );
+
+export const defaultDateRange = {
+	key: 'last_7_days',
+	date_start: timeOptions.last_7_days.start.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+	date_end: timeOptions.last_7_days.end.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+};
+
+export const DateRangePicker: React.FC< {
+	onDateRangeChange: ( dateRange: DateRange ) => void;
+	selectedKey: string;
+} > = ( { onDateRangeChange, selectedKey } ) => {
+	return (
+		<InlineLabelSelect
+			label="Select an option"
+			options={ options }
+			value={ options.find( ( option ) => option.key === selectedKey ) }
+			placeholder="Select an option..."
+			onChange={ ( changes ) => {
+				const selectedItem = changes.selectedItem;
+				if ( selectedItem ) {
+					const start = timeOptions[ selectedItem.key ].start.format(
+						'YYYY-MM-DD\\THH:mm:ss'
```

Comment:
Is it worthwhile using time values in tracks? The smallest resolution is a day so the time info is redundant. In the registration PR I'm assuming we'll send date only, e.g.

```js
start: YYYY-MM-DD, 
end: YYYY-MM-DD,
```

Thinking about this, these are not super useful props, we could remove them. For example, I don't think we'd want to group/segment by date, month or year. What do y'all think?

As an alternative, or maybe an additional prop, we could send total period days, which will allow us to segment data based on analysis period (e.g. month vs week vs year).

Reply:
Removing the precise date makes sense to me. The period would be [captured via](https://github.com/Automattic/woocommerce-payments/blob/bd95962f5ceff8a1df703db832c0c33dff284199/client/components/payment-activity/index.tsx#L136) `preset_name` ( renamed `key` )